board.h

```diff
@@ -0,0 +1,83 @@
+#pragma once
+#include <string>
+#include <vector>
+#include <fstream>
+
+// abstract parent class for both snakes and ladders
+// they just modify the positions differently, that's why I have used virtual function for it
+class Snake_Ladder
+{
+protected:
+    int low, high;
+
+public:
+    Snake_Ladder(int l, int h) : low(l), high(h) {}
+    virtual int modifyPosition(int pos) = 0;
+};
+
+class Snake : public Snake_Ladder
```

Comment:
Snake and Ladder classes should be in different files

Reply:
Okay, makes sense. Thanks!